lib/common/client/middleware/response/raise_error.rb

```diff
@@ -4,9 +4,49 @@ module Client
     module Middleware
       module Response
         class RaiseError < Faraday::Response::Middleware
+          attr_reader :service_acronym, :error_klass, :detail_key, :code_key, :source_key, :meta_key
+
+          def initialize(app, service_acronym, options = {})
+            @service_acronym = service_acronym
+            @error_klass = Common::Exceptions::BackendServiceException
+            @detail_key  = options[:detail_key]  || 'detail'
+            @code_key    = options[:code_key]    || 'code'
+            @source_key  = options[:source_key]  || 'source'
+            @meta_key    = options[:meta_key]    || 'meta'
+
+            super(app)
+          end
+
           def on_complete(env)
             return if env.success?
-            raise Common::Client::Errors::BackendServiceError.new(env.status.to_i, env[:body])
+            raise error_klass.new(i18n_error_key(env), response_values(env))
+          end
+
+          private
+
+          def code(env)
+            "#{service_acronym.upcase}#{env[:body][code_key]}"
+          end
+
+          def i18n_error_key(env)
+            if I18n.exists?("common.exceptions.#{code(env)}")
+              code(env)
+            else
+              Rails.logger.warn "The following exception should be added to locales: \n #{response_values(env)}"
+              'backend_service_exception'
+            end
+          end
+
+          # This method is not to be used for passing options. It is used to notify that
+          # a new error was presented from a backend service that is not currently mapped
+          # in locales
+          def response_values(env)
+            {
+              status: env.status.to_i,
+              detail: env[:body][detail_key],
+              code:   code(env),
+              source: env[:body][source_key]
```

Comment:
No, and you're write. I think a better strategy would be to have preliminary middleware before the raise_error that normalizes these and then the raise_error middleware just accepts argument for the prefix, 'Rx' or 'SM' I thought about introspecting somehow to get this as well, but couldn't think of a good way to do it.